src/core/mavlink_commands.cpp

```diff
@@ -278,39 +275,64 @@ void MavlinkCommandSender::receive_timeout()
 
 void MavlinkCommandSender::do_work()
 {
-    CommandResultCallback temp_callback = nullptr;
-    std::pair<Result, float> temp_result{Result::UnknownError, NAN};
+    LockedQueue<Work>::Guard work_queue_guard(_work_queue);
+    auto work = work_queue_guard.get_front();
+
+    if (!work) {
+        // Nothing to do.
+        return;
+    }
+
+    if (work->already_sent) {
+        return;
+    }
+
+    LogDebug() << "sending it the first time (" << work->mavlink_command << ")";
+    work->time_started = _parent.get_time().steady_time();
 
     {
-        LockedQueue<Work>::Guard work_queue_guard(_work_queue);
-        auto work = work_queue_guard.get_front();
+        std::lock_guard<std::mutex> lock(_sent_commands_mutex);
+        auto was_inserted =
+            _sent_commands.insert(std::make_pair(work->mavlink_command, work)).second;
```

Comment:
I'd probably use `const` as much as possible:

```suggestion
        const auto was_inserted =
            _sent_commands.insert(std::make_pair(work->mavlink_command, work)).second;
```

As well as below.